packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -716,6 +722,11 @@ DataGridPremiumRaw.propTypes = {
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onMenuOpen: PropTypes.func,
+  /**
+   * Callback fired when the pagination meta has changed.
+   * @param {GridPaginationMeta} paginationMeta Updated pagination meta.
+   */
+  onPaginationMetaChange: PropTypes.func,
```

Comment:
Not sure I understand this one, can pagination meta be changed by the data grid?

Reply:
The idea is to update the `paginationMeta` from the `dataSource` internally to handle index and cursor pagination with the three possibilities of the row count (known, unknown, and estimated)
So, not right now, but yes the Grid is likely to update the pagination meta internally when fetching data using the `dataSource` and store hasNextPage, nextCursor, and other information based on the currently active feature.
So the users could subscribe to `onPaginationMetaChange` if they want to get in sync with the internal state.

Does that make sense?

Reply:
Ah, got it. So the idea would be for the data source to provide the pagination meta, then the grid will update the paginationMeta state and call `onPaginationMetaChange`.
Makes sense to me 👍🏻